### DISCUSSION TITLE
How hard would it be to build an eslint plugin that does linting whenever code changes, much like the typescript typechecking plugin?

### DISCUSSION THREAD

**Discussion by karimb11** [microsoft/rnx-kit#2921] [Ideas]
Thanks!

**Comment by tido64** [comment]
Hi, I haven't looked into ESLint's API so I can't really tell how hard it would be. The TypeScript plugin is fairly complicated, but the Metro hook itself is not. You need to implement [`experimentalSerializerHook`](https://github.com/facebook/metro/blob/806fab3e7b95110e409ec91eacf59f279d4ec0e4/packages/metro-config/types/configTypes.d.ts#L127) in the Metro config. This will give you access to all the files being bundled. You can find our plugin's hook in `packages/metro-plugin-typescript/src/serializerHook.ts`:

https://github.com/microsoft/rnx-kit/blob/8083856287380695d4afb570d855480b6c5754bc/packages/metro-plugin-typescript/src/serializerHook.ts#L58

**Reply by karimb11** [reply]
Thank you 👍 , that hook was a pleasant surprise that I discovered in rnx-kit and it was the first thing that came to mind after reading through rnx code base, Metro documentation for it was sparse however.

I already tinkered with it by implementing a small plugin that auto clears the screen every time you save when running the development bundle server, it exists in Webpack and is quite handy. here it is if someone wants to use it, though I must apologize, it's quite rudimentary:

```
/* Inspired from: https://github.com/danillouz/clean-terminal-webpack-plugin/blob/master/index.js */

/* To avoid clearing Metro welcome screen. */
let firstRun = true;

function ClearTerminal() {
    return ( _entryPoint, _preModules, graph, _options ) => {
        if( firstRun ) {
            firstRun = false;
            return;
        }

        /* Skip in production. */
        if( process.env.NODE_ENV === "production" )
            return;

        /* Nothing changed, no need to reset anything. This mostly works right after the first
           run. */
        if(
            _preModules.added.size === 0 &&
            _preModules.modified.size === 0 &&
            _preModules.deleted.size === 0
        )
            return;

        process.stdout.write( "\x1B[2J\x1B[3J\x1B[H" );
    };
}

ClearTerminal.type = "serializerHook";

exports.ClearTerminal = ClearTerminal;
exports.default = ClearTerminal;

```

To use it, just save it in your repo as clear-terminal.js and add it to the rnx plugins list (Localized path)

**Reply by karimb11** [reply]
I didn't want to jump into an eslint plugin as I wasn't sure how feasible it is, like you, I also thought the TypeScript plugin seemed a bit on the heavy side, complexity wise.
